ee_plugin/utils.py

```diff
@@ -51,23 +50,7 @@ def add_ee_image_layer(image, name, shown, opacity):
     check_version()
 
     url = "type=xyz&url=" + get_ee_image_url(image)
-
-    # EE raster data provider
-    if image.ee_type == ee.Image:
-        layer = QgsRasterLayer(url, name, "EE")
-    # EE vector data provider
-    if image.ee_type in [ee.Geometry, ee.Feature]:
-        # TODO
-        layer = QgsRasterLayer(url, name, "wms")
-    # EE raster collection data provider
-    if image.ee_type == ee.ImageCollection:
-        # TODO
-        layer = QgsRasterLayer(url, name, "wms")
-    # EE vector collection data provider
-    if image.ee_type == ee.FeatureCollection:
-        # TODO
-        layer = QgsRasterLayer(url, name, "wms")
-
+    layer = QgsRasterLayer(url, name, "EE")
```

Comment:
Commenting for future 👀: So the practical impact here is that we are treating all Earth Engine data as a `QgsRasterLayer` layer registered with the "EE" provider which is managed via our custom `EarthEngineRasterDataProvider`.

https://github.com/gee-community/qgis-earthengine-plugin/blob/eb7a27e3c8838962eda57691db0092d2bbbebf9f/provider.py#L40-L68